lighthouse-core/audits/viewport.js

```diff
@@ -38,10 +39,29 @@ class Viewport extends Audit {
    * @return {!AuditResult}
    */
   static audit(artifacts) {
-    const hasMobileViewport = typeof artifacts.Viewport === 'string' &&
-        artifacts.Viewport.includes('width=');
+    if (typeof artifacts.Viewport !== 'string') {
+      return Viewport.generateAuditResult({
+        debugString: 'Error in determining viewport',
+        rawValue: -1
+      });
+    }
+
+    let debugString = '';
+    const parsedProps = Parser.parseMetaViewPortContent(artifacts.Viewport);
+
+    if (Object.keys(parsedProps.unknownProperties).length) {
+      debugString += `Invalid properties found: ${parsedProps.unknownProperties}`;
+    }
+    if (Object.keys(parsedProps.invalidValues).length) {
+      debugString += `Invalid values found: ${parsedProps.invalidValues}`;
```

Comment:
this debugString probably needs some separator between the two parts in case there are invalid properties and invalid values